app/validators/csv_validator.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CSVValidator < ActiveModel::Validator
+  def validate(csv_form)
+    @csv_form = csv_form
+
+    validate_file
+  end
+
+  private
+
+  attr_reader :csv_form
+
+  def file
+    csv_form.file
+  end
+
+  def validate_file
+    add_error :wrong_count unless valid_count?
+    add_error :wrong_type unless valid_extension?
+  end
+
+  def add_error(type)
+    csv_form.errors.add(:base, I18n.t("csv.validation.#{type}"))
+  end
+
+  def valid_count?
+    CSV.read(file).count.between?(1, 1000)
+  end
+
+  def valid_extension?
+    File.extname(file) == '.csv'
```

Comment:
```suggestion
    File.extname(file) == CSV_EXTENSION
```

we can declare it as a constant at the top of the class: `CSV_EXTENSION = '.csv'.freeze` ;-)